Sources/Services/Health/ServingStatus.swift

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// The status of a service.
+///
+/// - ``ServingStatus/serving`` indicates that a service is healthy.
+/// - ``ServingStatus/notServing`` indicates that a service is unhealthy.
+public struct ServingStatus: Sendable, Hashable {
+  package enum Value: Sendable, Hashable {
+    case serving
+    case notServing
+  }
+
+  /// A status indicating that a service is healthy.
+  public static let serving = ServingStatus(.serving)
+
+  /// A status indicating that a service unhealthy.
+  public static let notServing = ServingStatus(.notServing)
+
+  package var value: Value
```

Comment:
Do this prop and the `Value` enum really have to be `package`? Can't they be `private`?

Reply:
Oh yes, they don't have to be `package`. But they should be `internal`. The `value` property is accessed [in a different file](https://github.com/grpc/grpc-swift/pull/1987/files#diff-f379eb3d9b44ebf8b98c37c7f9af8ca6499e7500c83fc07c56c61ed4023cab09R120).